include/AutomationPattern.h

```diff
@@ -77,10 +79,11 @@ class LMMS_EXPORT AutomationPattern : public TrackContentObject
 	MidiTime timeMapLength() const;
 	void updateLength();
 
-	MidiTime putValue( const MidiTime & time,
+	MidiTime putValue(const MidiTime & time,
```

Comment:
It would feel cleaner to me to specify an inValue and outValue, instead of a value and outValueOffset. I know the current solution makes it easy to give the outValue a default value, but I'd rather use another solution:
- Hacky: set outValue's default value to NaN or infinity and handle that inside putValue
- Preferable IMO: Create a second putValue method that doesn't take an outValue, have it call the method that takes both and just repeat the inValue argument.

Reply:
I could try using two methods, but if I have `putValue(Miditime, float, bool, bool)` and `putValue(Miditime, float, float, bool, bool)` I'd have an ambiguity if I called `putValue(Miditime, float)` when just wanting to set the inValue right?

I could have the outValue at the last parameter instead, but I thought it would feel weird to have the inValue and outValue far apart in the arguments list so I ended up using offset. If you think it's fine to have them distant on the arguments list it should be an easy change!

Reply:
True, that is an issue. In that case how about these two:
- putValue(Miditime, float, bool, bool)
- putValue(Miditime, std::pair<float, float>, bool, bool)

This is completely unambiguous and keeps an intuitive argument order.

Reply:
Nice! That should work, I'll make the changes!

Reply:
We didn't think of the simplest solution:
`putValue` and `putValues` :laughing: 

I just pushed a commit with the changes, also changing the behavior of the drag node to keep the outValue constant if it's different from the inValue.